daemon/algod/server.go

```diff
@@ -172,9 +173,11 @@ func (s *Server) Start() {
 
 	s.stopping = make(chan struct{})
 
+	e := echo.New()
+
 	// use the data dir as the static file dir (for our API server), there's
 	// no need to separate the two yet. This lets us serve the swagger.json file.
-	apiHandler := apiServer.NewRouter(s.log, s.node, s.stopping, apiToken)
+	apiServer.ConfigureRouter(s.log, s.node, s.stopping, apiToken, e)
```

Comment:
Could you avoid configuring the default http mux but rather creating your own mux ?
( it's generally bad idea to change the default mux as there is only one of them. especially when you serve http requests on multiple ports.. )

Reply:
I think that echo is the separate mux. I actually had to take special steps to forward pprof endpoints to the default mux.

Is your suggestion that I move `echo.New()` into `ConfigureRouter` and rename it back to `NewRouter`?

Reply:
ok, so I think I got it. Echo replaces the handler in the provided server with it's own so it can process the incoming requests...
Yes, it would be nice to move it all into NewRouter, which would receive a tcpListener as well.
( I'm surprised there is no way to set the Handler on the http.Server to point to the echo handler.

Reply:
I made this change.

Also, it turns out `StartServer` on the echo object is a thin wrapper around `http.Server.Serve`, and sets itself to be the handler. So it's actually working exactly the way that you thought it should:
```
// StartServer starts a custom http server.
func (e *Echo) StartServer(s *http.Server) (err error) {
	// Setup
	e.colorer.SetOutput(e.Logger.Output())
	s.ErrorLog = e.StdLogger
	s.Handler = e
```